### PR TITLE
update modifiers appendix

### PR DESCRIPTION
**Pull request by nasbench**
- Updated the modifiers appendix with better titles and small description for each section.
- Added a new section to the `re` modifier to define the supported regex falvor and metacharacters.

### CODE REVIEW

File: appendix/sigma-modifiers-appendix.md [hunk 1]
```diff
@@ -78,6 +87,7 @@ The following document defines the standardized modifiers that can be used in Si
 
 ## History
 
+* 2024-08-10 Modifiers Appendix v2.0.0
```
Comment (frack113) [inline appendix/sigma-modifiers-appendix.md:90]:
```suggestion
* 2024-08-10 Modifiers Appendix v2.0.1
```

File: appendix/sigma-modifiers-appendix.md [hunk 2]
```diff
@@ -1,21 +1,23 @@
-# Modifiers <!-- omit in toc -->
+# Sigma Modifiers <!-- omit in toc -->
 
 The following document defines the standardized modifiers that can be used in Sigma.
 
-* Version 2.0.0
+* Version 2.0.1
 * Release date 2024-08-08
```
Comment (frack113) [inline appendix/sigma-modifiers-appendix.md:6]:
```suggestion
* Release date 2024-08-10
```

**Review by frack113 — APPROVED** [review]
Only typo change

**Comment by nasbench** [comment]
Made some additional updates to include section about supported regex falvor (to resolve #25)

File: appendix/sigma-modifiers-appendix.md [hunk 3]
```diff
@@ -62,11 +76,27 @@ The following document defines the standardized modifiers that can be used in Si
...
+* `lt`: Field is less than the value
+* `lte`: Field is less or equal than the value
+* `gt`: Field is greater than the value
+* `gte`: Field is greater or equal than the value
+
+### IP (Internet Protocol) Modifiers
+
+The modifiers listed in this section can only applied to IP values.
+
+* `cidr`: The value is handled as an CIDR by backends. Supports both IPv4 and IPv6 notations.
```
Comment (Res260) [inline appendix/sigma-modifiers-appendix.md:97]:
```suggestion
* `cidr`: The value is handled as an CIDR by backends. Supports both IPv4 and IPv6 notations. Example: `Source IP|cidr: 10.0.0.0/8`
```

File: appendix/sigma-modifiers-appendix.md [hunk 4]
```diff
@@ -32,28 +36,38 @@ The following document defines the standardized modifiers that can be used in Si
...
+    * Lookahead Assertions:
+      * Positive Lookahead: `(?=...)`
+      * Negative Lookahead: `(?!...)`
+      * Positive Lookbehind: `(?<=...)`
+      * Negative Lookbehind: `(?<!...)`
+    * Atomic Grouping: `(?>`
+
+
 * `re` sub-modifiers:
   * `i`: (insensitive) to enable case-sensitive matching.
```
Comment (nasbench) [inline appendix/sigma-modifiers-appendix.md:66]:
```suggestion
  * `i`: (insensitive) to enable case-insensitive matching.
```

**Pull request merged**